crates/bevy_reflect/src/lib.rs

```diff
@@ -1751,6 +1751,18 @@ mod tests {
         assert!(info.is::<MyValue>());
     }
 
+    #[test]
+    fn get_represented_kind_info() {
+        #[derive(Reflect)]
+        struct SomeStruct;
+
+        let ReflectRef::Struct(dyn_struct) = SomeStruct.reflect_ref() else {
+            unreachable!()
+        };
```

Comment:
Nit: you could probably simplify this
```suggestion
        let dyn_struct: &dyn Struct = &SomeStruct;
```